File: javaSrc/wms/model/Supply.java
package wms.model;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;

@Entity
@DiscriminatorValue(value = "supply")
public class Supply extends Invoice {
    private static final long serialVersionUID = 5011897484071966063L;

    public Supply() {
        super();
    }

}

File: application/web/app/model/entity/Supply.js
/*
 * Project: MajekWMS
 * User   : kornicameister
 * Date   : 04.11.12
 * Time   : 02:39
 */

Ext.define('WMS.model.entity.Supply', {
    extend: 'WMS.model.entity.Invoice',
    proxy : {
        type       : 'wms',
        url        : 'wms/agent/invoice',
        extraParams: {
            type: 'supply'
        }
    }
});